kernel/kernel-api/src/main/java/io/delta/kernel/internal/snapshot/LogSegment.java

```diff
@@ -186,6 +221,27 @@ public LogSegment(
                         Comparator.comparing((FileStatus a) -> new Path(a.getPath()).getName())
                             .reversed())
                     .collect(Collectors.toList()));
+
+    this.compactionsReversed =
+        new Lazy<>(
+            () ->
+                compactions.stream()
+                    .sorted(
+                        Comparator.comparing((FileStatus a) -> new Path(a.getPath()).getName())
```

Comment:
Can we sort by end version instead? Not lexicographically?

It would be more optimal, for example, to process the below in reverse endVersion order

```
80.100.json (start of the reverse list)
90.95.json
81.90.json
```

Lexicographical order should always be the endVersion order, except when an x.y has a bigger "range" (num commits inside it) than others around it.

Reply:
In general throughout our code in Kernel, we should probably clarify / specify exactly what "order" means (lexicographical vs numerical) :/ 

Another example, btw, is due to staged commit files, where the `_staged_commits` makes them come at the bottom of any sort order (which isn't great - we will need to fix this for catalogManaged tables)

Reply:
> perhaps if we assume constant size of compactions, then lexicographical order is equally as good? 

This was my assumption, but I realize it _could_ be different, say if different clients prefer different intervals. So I've switched to end version as it's equal or better (at minor increase in parsing cost)